Fix path resolution for local build on Windows

fearandesire opened the pull request:
Addresses the file pathing issue when attempting to compile or start a local build. 
Added `fileURLToPath` to convert the file URL to a file path string, which is compatible with both Unix and Windows systems.

I was able to run `yarn run build` and `yarn run start` on my local Windows and Linux operating systems successfully after the change.

dumbmatter commented:
This is nice!

1. Can you sign the CLA as described [here](https://github.com/zengm-games/zengm#license-and-contributor-license-agreement)?

2. Can you abstract this into a function, so it can be used in the other places __dirname is used currently? There are a few other places using it, if you search the codebase. Maybe create tools/lib/getDirname.js like:

```export const getDirname = (url: string) => {
    return = path.dirname(
		path.normalize(fileURLToPath(url)),
	);
}```

And then call it like:

```const __dirname = getDirname(import.meta.url);```

fearandesire commented:
> This is nice!
> 
>     1. Can you sign the CLA as described [here](https://github.com/zengm-games/zengm#license-and-contributor-license-agreement)?
> 
>     2. Can you abstract this into a function, so it can be used in the other places __dirname is used currently? There are a few other places using it, if you search the codebase. Maybe create tools/lib/getDirname.js like:
> 
> 
> ```
> export const getDirname = (url: string) => {
>     return = path.dirname(
> 		path.normalize(fileURLToPath(url)),
> 	);
> }
> ```
> 
> (Also see if that path.normalize call is really needed?)
> 
> And then call it like:
> 
> ```
> const __dirname = getDirname(import.meta.url);
> ```

No problem! Implemented those changes in the latest commit. I also removed the `path.normalize`, you were correct in the thought it wasn't needed.  

As for the CLA, I emailed a copy yesterday, 3/21/22 from (fenixcoding@gmail.com) - Would you want me to edit it on my repo as well? I thought I just had to send that email with the information filled out but definitely let me know otherwise/if I misread.

dumbmatter commented on tools/lib/watchJS.js:
```diff
@@ -1,7 +1,8 @@
+import { getDirname } from "./getDirname.js";
 
-const __dirname = path.dirname(new URL(import.meta.url).pathname);
+const __dirname = getDirname(new URL(import.meta.url));
```
Some calls to `getDirname` have `new URL()`, others don't. If both works, then leave it out.

fearandesire replied:
I took it out, and it works! Removed it in `getDirname` since the URLs being tunneled are objects.

dumbmatter commented on tools/races.js:
```diff
@@ -32,14 +33,14 @@ const defaultRaces: Record<string, Record<Race, number>> = ${JSON.stringify(
-	defaultRaces.Germany = defaultRaces.USA;
-	defaultRaces.Spain = defaultRaces.USA;
+    defaultRaces.USA = {
+        asian: 2,
+        black: 2,
+        brown: 3,
+        white: 93,
+    };
+    defaultRaces.Germany = defaultRaces.USA;
+    defaultRaces.Spain = defaultRaces.USA;
```
Not sure how this got changed, but prettier is supposed to be running on every commit, which would change this back to tabs. It is not a huge issue, but if you're going to be doing other PRs, it would be good to figure out if prettier is not working correctly, I guess it could be another Windows issue :)

fearandesire replied:
I checked the outputs for Prettier and thankfully it is linting properly, this was actually just a silly extension in my IDE setup. I've disabled it now and restored the formatting for this section

dumbmatter commented on tools/lib/getDirname.js:
```diff
@@ -0,0 +1,9 @@
+import { fileURLToPath } from "node:url";
+import path from "path";
```
Minor, but might as well make this "node:path"

fearandesire replied:
Done & done 😄

dumbmatter reviewed (changes requested):
Thanks! CLA was in my spam folder :)

Couple little comments and then I think it'll be done.

dumbmatter commented:
Thanks!
